src/main/java/org/example/services/JobRoleService.java

```diff
@@ -19,10 +20,19 @@ public JobRoleService(final JobRoleDao jobRoleDao,
 
     public List<JobRoleResponse> getAllJobRoles() throws SQLException {
         try {
-            return jobRoleMapper.mapOrderListToJobRoleResponseList(
+            return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
                     jobRoleDao.getJobRoles());
         } catch (SQLException e) {
             throw new SQLException(e);
         }
     }
+    public List<JobRoleDetailedResponse> getJobRoleById(final int id)
+            throws SQLException {
+        try {
+            return jobRoleMapper.mapJobRoleListToJobRoleDetailedResponse(
+                    jobRoleDao.getJobRoleById(id));
+        } catch (SQLException e) {
+            throw new SQLException(e);
```

Comment:
Because you are throwing it in the signature here Luke, you dont need to catch and rethrow it... it will simply get thrown up automatically